homeassistant/components/honeywell/climate.py

```diff
@@ -111,17 +111,25 @@ def remove_stale_devices(
     device_entries = dr.async_entries_for_config_entry(
         device_registry, config_entry.entry_id
     )
-    all_device_ids: list = []
+    all_device_ids: set = set()
     for device in devices.values():
-        all_device_ids.append(device.deviceid)
+        all_device_ids.add(device.deviceid)
 
     for device_entry in device_entries:
         device_id: str | None = None
+        dont_remove: bool = False
 
         for identifier in device_entry.identifiers:
+            if identifier[0] != DOMAIN:
+                dont_remove = True
+                continue
+
             device_id = identifier[1]
             break
 
+        if dont_remove is True:
+            continue
```

Comment:
Alternatively we could include the new boolean in the condition below on line 133.

Reply:
"Duh" - makes perfect sense :)